### ISSUE TITLE
Automatic equation numbering when referencing with :ref:

### ISSUE THREAD

**Issue by benRenard** [DassHydro/smash#127]
In the Sphynx doc, the equation number in references to equations is hard-coded, e.g. :
The likelihood function in :ref:\`equation 3 <math_num_documentation.bayesian_estimation.likelihood>\`

It would be nice to be able to use automatic numbering, e.g. something that could look like:
The likelihood function in :ref:\`equation %s <math_num_documentation.bayesian_estimation.likelihood>\` 

Apparently it's doable by playing with the numref/numfig/etc. options in conf.py.

**Comment by nghi-truyen** [comment]
Yes, this can be done by writing for example Eq. :eq:`math_num_documentation.bayesian_estimation.likelihood` instead of using :ref:
I will correct all of them in the next PR.
